source/getEntityTypescriptPostgres.ts

```diff
@@ -1,169 +1,522 @@
-import {createPostgresDataProvider} from 'remult/postgres';
-import {writeFile, mkdir} from 'fs/promises';
+import {
+	existsSync,
+	lstatSync,
+	mkdirSync,
+	readdirSync,
+	rmdirSync,
+	unlinkSync,
+	writeFileSync,
+} from 'fs';
+import {green, yellow} from 'kleur/colors';
+import {SqlDatabase, remult} from 'remult';
 
-export function generateModels(
-	connectionString: string,
-	tables: string[],
-	dir = '.',
+const deleteFolderRecursive = path => {
+	if (existsSync(path)) {
+		readdirSync(path).forEach(file => {
+			const curPath = `${path}/${file}`;
+			if (lstatSync(curPath).isDirectory()) {
+				// Recursive call for subdirectories
+				deleteFolderRecursive(curPath);
+			} else {
+				// Delete file
+				unlinkSync(curPath);
+			}
+		});
+		// Delete empty directory
+		rmdirSync(path);
+	} else {
+	}
+};
+
+const toPascalCase = (str: string) => {
+	return str
+		.replace(/[-_](\w)/g, (_, c) => c.toUpperCase())
+		.replace(/^\w/, c => c.toUpperCase());
+};
+
+function toCamelCase(str) {
+	let words = toPascalCase(str).split('');
+	words[0] = words[0].toLowerCase();
+	return words.join('');
+}
+
+const toTitleCase = str => {
+	return str
+		.toLowerCase()
+		.replace(/_/g, ' ')
+		.replace(/\w+/g, match => match.charAt(0).toUpperCase() + match.slice(1));
+};
+
+function build_column(
+	decorator: string,
+	decoratorArgsValueType: string,
+	decoratorArgsOptions: string[],
+	column_name_tweak: string,
+	column_name: any,
+	is_nullable: 'YES' | 'NO',
+	type: string,
+	defaultVal: string,
 ) {
-	mkdir(dir, {recursive: true});
-	Promise.all(
-		tables.map(async table => {
-			const entity = await getEntityTypescriptPostgres(table, connectionString);
-			writeFile(`${dir}/${table}.ts`, entity, {flag: 'w'});
-		}),
-	);
+	if (
+		column_name.toLocaleLowerCase() !== column_name ||
+		column_name_tweak ||
+		column_name === 'order'
+	) {
+		decoratorArgsOptions.unshift(`dbName: '"${column_name}"'`);
+	}
+
+	if (is_nullable === 'YES') {
+		decoratorArgsOptions.push(`allowNull: true`);
+	}
+
+	let decoratorArgs = [];
+	if (decoratorArgsValueType) {
+		decoratorArgs.push(decoratorArgsValueType);
+	}
+
+	if (decoratorArgsOptions.length > 0) {
+		decoratorArgs.push(`{ ${decoratorArgsOptions.join(', ')} }`);
+	}
+
+	let current_col = `\t${decorator}(${decoratorArgs.join(', ')})\n\t${
+		column_name_tweak ? column_name_tweak : column_name
+	}`;
+
+	if (is_nullable === 'YES') {
+		current_col += '?';
+	}
+	if (is_nullable !== 'YES' && !defaultVal) {
+		current_col += '!';
+	}
+
+	if (type) {
+		current_col += ': ';
+		current_col += type;
+	}
+	if (defaultVal) {
+		current_col += ' = ' + defaultVal;
+	}
+
+	return current_col;
+}
+
+class DbTable {
+	schema: string;
+	db_name: string;
+	graphql_key: string;
```

Comment:
i think we should not tie it to graphql it should be entity like remult semantics

Reply:
You are right, the naming is not correct. It's called "key" in remult, it's this param: 
![image](https://github.com/Yedidyar/remult-cli/assets/5312607/f9eb140e-af05-4972-8ca3-95023d8bd2e8)

`key` is probably good enough ✌️